baseplate/clients/thrift.py

```diff
@@ -223,119 +264,150 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
                     span.set_tag("method", method.__name__)
                     span.start()
 
-                    try:
-                        baseplate = span.baseplate
-                        if baseplate:
-                            service_name = baseplate.service_name
-                            if service_name:
-                                prot.trans.set_header(b"User-Agent", service_name.encode())
-
-                        prot.trans.set_header(b"Trace", str(span.trace_id).encode())
-                        prot.trans.set_header(b"Parent", str(span.parent_id).encode())
-                        prot.trans.set_header(b"Span", str(span.id).encode())
-                        if span.sampled is not None:
-                            sampled = "1" if span.sampled else "0"
-                            prot.trans.set_header(b"Sampled", sampled.encode())
-                        if span.flags:
-                            prot.trans.set_header(b"Flags", str(span.flags).encode())
-
-                        min_timeout = time_remaining
-                        if self.pool.timeout:
-                            if not min_timeout or self.pool.timeout < min_timeout:
-                                min_timeout = self.pool.timeout
-                        if min_timeout and min_timeout > 0:
-                            # min_timeout is in float seconds, we are converting to int milliseconds
-                            # rounding up here.
-                            prot.trans.set_header(
-                                b"Deadline-Budget", str(int(ceil(min_timeout * 1000))).encode()
-                            )
-
+                    mutable_metadata: OrderedDict = OrderedDict()
+
+                    pool_addr = self.pool.endpoint.address
+                    if isinstance(pool_addr, str):
+                        otel_attributes[SpanAttributes.NET_PEER_IP] = pool_addr
+                    elif pool_addr is not None:
+                        otel_attributes[SpanAttributes.NET_PEER_IP] = pool_addr.host
+                        otel_attributes[SpanAttributes.NET_PEER_PORT] = pool_addr.port
+                    if otel_attributes.get(SpanAttributes.NET_PEER_IP) in ["127.0.0.1", "::1"]:
+                        otel_attributes[SpanAttributes.NET_PEER_NAME] = "localhost"
```

Comment:
When does localhost traffic like this happen? For the IPv4 address I guess in theory it could be all of 127.0.0.0/8 depending on how this is used? Mostly wondering because it's not uncommon to see e.g. `127.0.1.1` used in default Linux setups in `/etc/hosts`.

Reply:
Honestly I think the effort here is to just provide a default `NET_PEER_NAME` for some common scenarios. It's kind of a convenience if you are filtering or grouping by hostnames I guess so that you get a value instead of empty.